[WebAuthn] Update rpId type to DOMString

pascoej opened the pull request:
#### b9e93c034a7e55038d1fd708fdc3af8c1bbf164e
<pre>
[WebAuthn] Update rpId type to DOMString
<a href="https://bugs.webkit.org/show_bug.cgi?id=277439">https://bugs.webkit.org/show_bug.cgi?id=277439</a>
<a href="https://rdar.apple.com/132918362">rdar://132918362</a>

Reviewed by Brent Fulgham.

This is to address a small, nit-like change in the spec to change the type of rpId in
PublicKeyCredentialRequestOptions to be DOMString instead of USVString to match
the member in PublicKeyCredentialRpEntity.id.

See <a href="https://github.com/w3c/webauthn/pull/2098.">https://github.com/w3c/webauthn/pull/2098.</a>

* Source/WebCore/Modules/webauthn/PublicKeyCredentialRequestOptions.idl:

Canonical link: <a href="https://commits.webkit.org/281674@main">https://commits.webkit.org/281674@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/26ed0346601095396e58b94c1c0374ffe7fe8348

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/60603 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/39962 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/13179 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/64534 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/11150 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/62733 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/47638 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/11439 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/49025 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/7747 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/62636 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/37224 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/52503 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/29853 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/33915 "Passed tests") | [❌ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/9732 "Found 1 new test failure: imported/w3c/web-platform-tests/content-security-policy/reporting-api/reporting-api-report-only-sends-reports-on-violation.https.sub.html (failure)") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/10063 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/55779 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/10033 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/66263 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/4547 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/62/builds/9900 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/56391 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/4568 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/52477 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/56564 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/13507 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/3773 "Passed tests") | | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/35767 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/36849 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/37942 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/36594 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

brentfulgham commented on Source/WebCore/Modules/webauthn/PublicKeyCredentialRequestOptions.idl:
```diff
@@ -28,7 +28,7 @@
     required BufferSource challenge;
     unsigned long timeout;
-    USVString rpId;
+    DOMString rpId;
```
Does this have any web-visible impact? How do we know this didn't break any existing credential request flows (e.g., is it validated by existing test cases?)

pascoej replied:
I'm confident this doesn't have any web-visible impact because the WebAuthn spec has already has limited PublicKeyCredentialRequestOptions.rpId to a valid domain, which is more restrictive than a USVString (full reasoning in https://github.com/w3c/webauthn/pull/2098.) We also already take rpId as a DOMString in another method.

brentfulgham reviewed (approved):
Looks fine, as long as we are confident it doesn't regress web-exposed use cases.

pascoej committed:
[WebAuthn] Update rpId type to DOMString
https://bugs.webkit.org/show_bug.cgi?id=277439
rdar://132918362

Reviewed by Brent Fulgham.

This is to address a small, nit-like change in the spec to change the type of rpId in
PublicKeyCredentialRequestOptions to be DOMString instead of USVString to match
the member in PublicKeyCredentialRpEntity.id.

See https://github.com/w3c/webauthn/pull/2098.

* Source/WebCore/Modules/webauthn/PublicKeyCredentialRequestOptions.idl:

Canonical link: https://commits.webkit.org/281674@main
